Sources/MeiliSearch/Model/SearchParameters.swift

```diff
@@ -72,35 +83,60 @@ public struct SearchParameters: Codable, Equatable {
     }
 
     func dictionary() -> [String: String] {
-      var dic = [String: String]()
-      dic["q"] = query
-      dic["offset"] = "\(offset)"
-      dic["limit"] = "\(limit)"
-
-      if let attributesToRetrieve = self.attributesToRetrieve {
-        dic["attributesToRetrieve"] = commaRepresentation(attributesToRetrieve)
-      }
-
-      if !attributesToCrop.isEmpty {
-        dic["attributesToCrop"] = commaRepresentation(attributesToCrop)
-      }
-
-      dic["cropLength"] = "\(cropLength)"
-
-      if !attributesToHighlight.isEmpty {
-        dic["attributesToHighlight"] = commaRepresentation(attributesToHighlight)
-      }
-
-      if let filters: Filter = self.filters {
-        dic["filters"] = "\(filters.attribute):\(filters.value)"
-      }
-
-      dic["matches"] = "\(matches)"
-      return dic
+        var dic = [String: String]()
+        dic["q"] = query
+        dic["offset"] = "\(offset)"
+        dic["limit"] = "\(limit)"
+
+        if let attributesToRetrieve = self.attributesToRetrieve {
+            dic["attributesToRetrieve"] = commaRepresentation(attributesToRetrieve)
+        }
+
+        if !attributesToCrop.isEmpty {
+            dic["attributesToCrop"] = commaRepresentation(attributesToCrop)
+        }
+
+        dic["cropLength"] = "\(cropLength)"
+
+        if !attributesToHighlight.isEmpty {
+            dic["attributesToHighlight"] = commaRepresentation(attributesToHighlight)
+        }
+
+        if let filters: Filter = self.filters {
+            dic["filters"] = "\(filters.attribute):\(filters.value)"
+        }
+
+        if let facetFilters: String = self.facetFilters {
+  //        var facets: [String] = []
+  //        facetFilters.forEach { any in
+  //          switch any {
+  //          case let filter as Filter:
+  //            print("ddd")
+  ////            facets
+  //          default:
+  //            print("ddsssd")
+  //          }
+  //        }
```

Comment:
```suggestion
```

Reply:
I left this because I was not sure how the client should handle the faceted filters. I wanted to make it as `[Any]` and use casting inside the function to map the values but since `Codable` does not support `Any` mapping, I am unable to do this. The only way I can see to do this to have a type that accepts either `String` and `Tree`. I just find `String` faster for the moment. Any opinion?

Reply:
What you mean is you cannot handle the `[String]` and the `[[String]]` in the same variable?
Ex: `['genre:romance', [genre:action]]`

Because it's impossible to do that in Goland too, we chose to enforce the `[[String]]` everywhere and we will update the docs accordingly.
What I mean is the user must write 
`[['genre:romance'], [genre:action]]` and not `['genre:romance', [genre:action]]`.
Indeed, `[['genre:romance']] == ['genre:romance']` for MeiliSearch

Am I wrong @eskombro?

Sorry @ppamorim if I completely miss understood your issue 😅

Reply:
If multiple typing is possible in Swift this is what I did in typescript: 


```javascript
// types.ts
export type FacetFilter = string | FacetFilter[]

export interface SearchParams {
  offset?: number
  limit?: number
  attributesToRetrieve?: string[] | string
  attributesToCrop?: string[] | string
  cropLength?: number
  attributesToHighlight?: string[] | string
  filters?: string
  facetFilters?: FacetFilter[]
  facetsDistribution?: string[]
  matches?: boolean
}

export interface SearchRequest {
  q: string
  offset?: number
  limit?: number
  attributesToRetrieve?: string
  attributesToCrop?: string
  cropLength?: number
  attributesToHighlight?: string
  facetFilters?: string
  facetsDistribution?: string
  filters?: string
  matches?: boolean
}
```

- `SearchParams` is the `options` parameter of the search function.
- `SearchRequest` is the query parameters that I send to MeiliSearch.

In searchParams facetFilters is composed of an array of facetFilters. 
`facetFilters?: FacetFilter[]`
That is declared like this
`export type FacetFilter = string | FacetFilter[]`

This recursive typing makes it possible to have any deepness of array and string. Thus making the following possible: 
`[["genre:comedy"], "genre:action"]`.

Now the problem is that MeiliSearch only accepts a `string` as input value. Which is why in the `search` method I need to transform `facetFilter` into a string. 
This is done like this:
```javascript
params.facetFilters = JSON.stringify(options.facetFilters) // Here I stringify 
```
`params` in this example is of type `SearchRequest`.

I have no idea is this is of any use for you. I hope it is.

Reply:
Multiple typing is not possible in Swift so far I know.